Added CTkTooltip

supercam19 opened the pull request:
I have added tooltips which appear when your mouse hovers over a widget. These can be very useful for describing the purposes of widgets such as buttons, without using the widget's text to describe it. This is possible using Toplevels and transparent colors to give the tooltip rounded edges.

![image](https://user-images.githubusercontent.com/79729769/216791350-c87c92e4-c861-41cf-ba79-ee71ed870574.png)


Here is a more practical example, where tooltips are used to indicate the usage of a button represented only with an icon:
![image](https://user-images.githubusercontent.com/79729769/213933911-5da900eb-6bfb-4a48-a9be-1ae65706574a.png)
Allowing for a clean and modern user interface.

Please consider adding this as I think many people would find it useful

supercam19 committed:
Add CTkTooltip

supercam19 committed:
Pass kwargs into label instead of toplevel

supercam19 committed:
Add a few comments

TomSchimansky commented:
The implementation looks good, but the problem is that it does not work on all platforms, for example on macOS in fullscreen mode, it would open a new tab, because you cannot open a secondary window in fullscreen mode above the existing one there.

UserJoo9 commented:
awesome, am very interest !!!

supercam19 committed:
Fix MacOS error; color changes

Fixed errors using MacOS trying to create a tooltip

supercam19 commented:
I've tested tooltips in MacOS fullscreen and they seem to work as expected.

Converting PR to a draft to fix some differences across platforms

supercam19 committed:
Fix issues caused by slow machines

Wolf-SO commented:
I see accessibility issues with the color scheme shown. In general, tooltips seem to be designed to stand out visually. So what about inverting colors?

supercam19 commented:
@Wolf-SO you're right, I hadn't updated the screenshot yet. The image has now been changed to show the higher contrast to the default colors of the theme.

IshanJ25 commented:
nice

can it track mouse movement to update location in real-time?

supercam19 committed:
Tooltip improvements

- Fix .configure() not working
- Fix tooltips sometimes not disappearing
- Add is_visible() method to return visibility status

supercam19 committed:
Remove debug

supercam19 committed:
Fix multiple tooltips sometimes appearing at once

supercam19 commented:
@IshanJ25 It's not built in to the tooltip, but it can still be done. Ex:
```python
import customtkinter
import threading
import time

app = customtkinter.CTk()

def track_mouse():
    tooltip_1.configure(delay=0)
    while app.winfo_exists():
        if tooltip_1.is_visible():
            tooltip_1.geometry(f'{app.winfo_pointerx()+1}+{app.winfo_pointery()+1}')
            time.sleep(0.02)

frame_1 = customtkinter.CTkFrame(master=app)
frame_1.pack(pady=20, padx=60, fill="both", expand=True)

button_1 = customtkinter.CTkButton(master=frame_1)
button_1.pack(pady=10, padx=10)
tooltip_1 = customtkinter.CTkTooltip(master=button_1)
tooltip_follow_thread = threading.Thread(target=track_mouse)
tooltip_follow_thread.start()

app.mainloop()
```

FedericoSpada commented:
Hi,
I've developed the same widget myself for a project.
I've tried to replicate the look of the Microsoft Office ToolTips, with a short title and a description:
<img width="232" height="127" alt="Screenshot_2026 01 07_08 51 15" src="https://github.com/user-attachments/assets/f4024fee-b6ea-42f8-8e55-15b26e47cb8a" />
It can also update its content dynamically.

The difference is that I've used a CTkFrame properly placed instead of a CTkTopLevel. I plan to make it available in a next release.
For now, I leave this PR open so that it is more searchable if someone in the meantime wants to use this one.

ElectricCandlelight commented:
Akascape has a really nice tooltip, worth a look. [CTkToolTip](https://github.com/Akascape/CTkToolTip)

FedericoSpada commented:
Coincidentally, someone else just pointed out the same widget in #2254, and it turned out that it is mixing Tk and CTk widgets, resulting in an Exception.
I'll have a look if it has additional useful features that are missing from both this version and mine, and I'll try to integrate them.

ajkessel commented:
Any movement on this? I've been using [CTkToolTip](https://github.com/Akascape/CTkToolTip/) but am experiencing the `AttributeError: '_SizedToolTip' object has no attribute 'block_update_dimensions_event'` bug. I don't know if the more practical approach is to try to fix this in CTkTooltip or wait for this PR to be merged into customtkinter.

FedericoSpada commented:
Hi all,
I've finally implemented the CTkToolTip widget in the new [repository](https://github.com/FedericoSpada/Custom2kinter) that I created to continue the development.
My plan is to release a new version on PyPI in 1 or 2 months. Consider that the name has changed to [Custom2kinter](https://pypi.org/project/custom2kinter/), as reported in this [discussion](https://github.com/TomSchimansky/CustomTkinter/discussions/2343#discussioncomment-16532895).
